exporters/zipkin/include/opentelemetry/exporters/zipkin/zipkin_exporter.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "opentelemetry/exporters/zipkin/recordable.h"
+#include "opentelemetry/ext/http/client/http_client_factory.h"
+#include "opentelemetry/ext/http/common/url_parser.h"
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include "nlohmann/json.hpp"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace zipkin
+{
+
+std::string ZIPKIN_ENDPOINT_DEFAULT = "http://localhost:9411/api/v2/spans";
+
+/**
+ * Struct to hold Zipkin  exporter options.
+ */
+struct ZipkinExporterOptions
+{
+  // The endpoint to export to. By default the OpenTelemetry Collector's default endpoint.
+  std::string endpoint     = ZIPKIN_ENDPOINT_DEFAULT;
+  TransportFormat format   = TransportFormat::JSON;
+  std::string service_name = "default-service";
+  std::string ipv4;
+  std::string ipv6;
+};
+
+namespace trace_sdk   = opentelemetry::sdk::trace;
+namespace http_client = opentelemetry::ext::http::client;
+
+/**
+ * The Zipkin exporter exports span data in JSON format as expected by Zipkin
+ */
+class ZipkinExporter final : public trace_sdk::SpanExporter, public http_client::EventHandler
```

Comment:
Why do you derive from `EventHandler` here?

Reply:
Good catch. It is not needed, and so removed.